hathor/p2p/node_sync_v2.py

```diff
@@ -0,0 +1,1331 @@
+        data = json.loads(payload)
+        assert self.protocol.connections is not None
+        self.log.debug('got best block', **data)
+        data['block'] = bytes.fromhex(data['block'])
+
+        key = 'best-block'
+        deferred = self.deferred_by_key.pop(key, None)
+        if deferred:
+            deferred.callback(data)
+
+    def _setup_tx_streaming(self):
+        self._tx_received = 0
+        self._tx_max_quantity = DEAFAULT_STREAMING_LIMIT  # XXX: maybe this is redundant
+        # XXX: what else can we add for checking if everything is going well?
+
+    def send_get_block_txs(self, child_hash: bytes, last_block_hash: bytes) -> None:
+        """ Request a BFS of all transactions that parent of CHILD, up to the ones first comfirmed by LAST-BLOCK.
+
+        Note that CHILD can either be a block or a transaction. But LAST-BLOCK is always a block.
+        """
+        self._setup_tx_streaming()
+        self.log.debug('send_get_block_txs', child=child_hash.hex(), last_block=last_block_hash.hex())
+        payload = json.dumps(dict(
+            child=child_hash.hex(),
+            last_block=last_block_hash.hex(),
+        ))
+        self.send_message(ProtocolMessages.GET_BLOCK_TXS, payload)
+        self.receiving_stream = True
+
+    def handle_get_block_txs(self, payload: str) -> None:
+        if self._is_streaming:
+            self.log.warn('already streaming')
+            # self.log.warn('ignore GET-BLOCK-TXS, already streaming')
+            # return
+        data = json.loads(payload)
+        self.log.debug('handle_get_block_txs', **data)
+        child_hash = bytes.fromhex(data['child'])
+        last_block_hash = bytes.fromhex(data['last_block'])
+        self.send_block_txs(child_hash, last_block_hash)
+
+    def send_block_txs(self, child_hash: bytes, last_block_hash: bytes) -> None:
+        try:
+            tx = self.tx_storage.get_transaction(child_hash)
+        except TransactionDoesNotExist:
+            # In case the tx does not exist we send a NOT-FOUND message
+            self.send_message(ProtocolMessages.NOT_FOUND, child_hash.hex())
+            return
+        if not self.tx_storage.transaction_exists(last_block_hash):
+            # In case the tx does not exist we send a NOT-FOUND message
+            self.send_message(ProtocolMessages.NOT_FOUND, last_block_hash.hex())
+            return
+        x = TransactionsStreaming(self, tx, last_block_hash)
+        x.start()
+
+    def send_transaction(self, tx: Transaction) -> None:
+        """Send a TRANSACTION message."""
+        # payload = bytes(tx).hex()  # fails for big transactions
+        payload = base64.b64encode(bytes(tx)).decode('ascii')
+        self.send_message(ProtocolMessages.TRANSACTION, payload)
+
+    def handle_transaction(self, payload: str) -> None:
+        """Handle a received TRANSACTION message."""
+        assert self.protocol.connections is not None
+
+        # if self.state != PeerState.SYNCING_TXS:
```

Comment:
Should we remove this commented piece of code?

Reply:
Removed.